contracts/ERC1948.sol

```diff
@@ -52,7 +52,55 @@ contract ERC1948 is IERC1948, ERC721 {
    * - `msg.sender` needs to be owner of `tokenId`.
    */
   function writeData(uint256 tokenId, bytes32 newData) external {
-    require(msg.sender == ownerOf(tokenId));
+    require(msg.sender == ownerOf(tokenId) || getApproved(tokenId) == msg.sender, "no permission");
+    emit DataUpdated(tokenId, data[tokenId], newData);
+    data[tokenId] = newData;
+  }
+
+
+  function _ecRecoverPersonal(bytes32 _before, bytes32 _after, bytes memory signature) internal pure returns (address) {
+    // Check the signature length
+    if (signature.length != 65) {
+      return (address(0));
+    }
+
+    // Divide the signature in r, s and v variables
+    bytes32 r;
+    bytes32 s;
+    uint8 v;
+
+    // ecrecover takes the signature parameters, and the only way to get them
+    // currently is to use assembly.
+    // solhint-disable-next-line no-inline-assembly
+    assembly {
+      r := mload(add(signature, 0x20))
```

Comment:
You need to additionally add 0x20 to the signature ptr because first word is length of the bytes array

Reply:
Nevermind, I should have read the other parts first :)